src/main/java/org/jabref/gui/ReplaceStringView.java

```diff
@@ -0,0 +1,83 @@
+package org.jabref.gui;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.TextField;
+import javafx.stage.Stage;
+
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.gui.util.IconValidationDecorator;
+import org.jabref.logic.l10n.Localization;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+
+public class ReplaceStringView extends BaseDialog<Void>
+{
+
+    @FXML private ButtonType replaceButton;
+    @FXML private TextField limitFieldInput;
+    @FXML private TextField findField;
+    @FXML private TextField replaceField;
+    @FXML private DialogPane pane;
+
+    private boolean allFieldReplace;
+    private boolean selOnly;
+    private BasePanel panel;
+    private Stage stage;
+
+    private final ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
+
+    public ReplaceStringView(BasePanel basePanel)  {
+        this.setTitle(Localization.lang("Replace String"));
+
+        allFieldReplace = true;
+        selOnly = false;
+        panel = basePanel;
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        stage = (Stage) this.pane.getScene().getWindow();
+        ControlHelper.setAction(replaceButton, getDialogPane(), event -> buttonReplace());
+    }
+
+    @FXML
+    public void initialize() {
+        visualizer.setDecoration(new IconValidationDecorator());
+    }
+
+    @FXML
+    public void buttonReplace() {
+        String findString = findField.getText();
+        String replaceString = replaceField.getText();
+        String[] fieldStrings = limitFieldInput.getText().toLowerCase().split(";");
+        if ("".equals(findString))
+        {
+            stage.close();
+            return;
+        }
+        ReplaceStringViewModel viewModel = new ReplaceStringViewModel(panel, fieldStrings, findString, replaceString, selOnly, allFieldReplace);
```

Comment:
This works because the dialog is very simple, but goes against the usual strategy of JavaFX / MVVM. You should add properties in the ViewModel class and bind these properties to the corresponding properties of the controls. Please have a look at the other JavaFX dialogs to see how this is done. 

Since this approach is a bit different from what you are used to in the Swing world, let me quickly explain the big overall picture and its advantages. You are probably used to program against the UI controls (as you do here right now): for example, if you want to know the input from the user, you use `textbox.getText()` to get it. Similarly, for other input controls. This works all nicely until the user reports a bug and you have to write a test for it. The problem you then face is the following: how do I automate the user interaction in my test? There are different approaches to this issue but the general answer is: there is no good solution ;-) UI automation is usually ugly and breaks as soon as you change the design. The idea behind the MVVM approach is that you completely separate the UI from the logic that performs the action. So you create a class (the ViewModel) that holds the state of the UI (i.e. you have `string` field that corresponds to the user input, but at this point you don't care how or where this string is manipulated). This state in the ViewModel is then bound to the actual userinterface in the `View` class using JavaFX bindings. In this way, you can test the ViewModel, i.e. the complete logic, without ever knowing how the user interface looks like.